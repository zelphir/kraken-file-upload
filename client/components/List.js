import React from 'react'
import PropTypes from 'prop-types'
import { Button, NonIdealState, Tag } from '@blueprintjs/core'
import { getClassNameForExtension } from 'font-awesome-filetypes'
import DownloadButton from './DownloadButton'
import UploadButton from './UploadButton'

const List = ({ data, onDelete, onUpload, hasQuery }) => {
  if (!data.length && hasQuery)
    return <NonIdealState visual="search" title="No result!" className="spacer" />

  return !data.length ? (
    <NonIdealState visual="folder-open" title="Your folder is empty!" className="spacer">
      <UploadButton large onUpload={onUpload} />
    </NonIdealState>
  ) : (
    <table className="pt-html-table pt-html-table-striped">
      <thead>
        <tr>
          <th>File</th>
          <th style={{ width: 100 }}>Actions</th>
        </tr>
      </thead>
      <tbody>
        {data.map(file => (
          <tr key={file._id}>
            <td>
              <i className={`fa ${getClassNameForExtension(file.type)} fa-lg`} />
              <DownloadButton className="filename" filename={file.filename}>
                {file.originalName}
              </DownloadButton>
              <Tag minimal>
                <em>{Math.round(file.size * 100) / 100}kb</em>
              </Tag>
            </td>
            <td>
              <DownloadButton icon="download" intent="primary" filename={file.filename} />
              <Button
                minimal
                className="delete"
                icon="delete"
                intent="danger"
                onClick={onDelete}
                value={file.filename}
              >
                {file.filename}
              </Button>
            </td>
          </tr>
        ))}
      </tbody>
    </table>
  )
}

List.propTypes = {
  data: PropTypes.array.isRequired,
  onDelete: PropTypes.func.isRequired,
  onUpload: PropTypes.func.isRequired,
  hasQuery: PropTypes.bool.isRequired
}

export default List
