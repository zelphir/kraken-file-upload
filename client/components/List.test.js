import React from 'react'
import { shallow } from 'enzyme'
import List from './List'

function setup(nextProps = {}) {
  const props = {
    data: [],
    onDelete: jest.fn(),
    onUpload: jest.fn(),
    ...nextProps
  }
  const wrapper = shallow(<List {...props} />)

  return {
    props,
    wrapper
  }
}

describe('<List />', () => {
  it('should match the snapshot', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the empty message if no data', () => {
    const { wrapper } = setup()
    expect(wrapper.find('NonIdealState')).toHaveLength(1)
  })

  it('should render the table if data', () => {
    const { wrapper } = setup({
      data: [{ _id: 1, filename: 'a.txt', originalName: 'b.txt', type: 'txt', size: 1 }]
    })
    expect(wrapper.find('NonIdealState')).toHaveLength(0)
    expect(wrapper.find('table')).toHaveLength(1)
  })
})
