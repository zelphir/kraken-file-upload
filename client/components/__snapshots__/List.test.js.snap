// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should match the snapshot 1`] = `
<NonIdealState
  className="spacer"
  title="Your folder is empty!"
  visual="folder-open"
>
  <UploadButton
    large={true}
    onUpload={[MockFunction]}
  />
</NonIdealState>
`;
